test/server/cards/03_TWI/events/GrievousReassembly.spec.ts

```diff
@@ -0,0 +1,54 @@
+describe('Grievous Reassembly', function() {
+    integration(function(contextRef) {
+        describe('Grievous Reassembly\'s ability', function() {
+            it('should heal 3 from a unit and create a Battle Droid token', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['grievous-reassembly'],
+                        groundArena: [{ card: 'fifth-brother#fear-hunter', damage: 3 }]
+                    },
+                    player2: {
+                        groundArena: ['atst']
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.grievousReassembly);
+                expect(context.player1).toBeAbleToSelectExactly([context.fifthBrother, context.atst]);
+
+                // Selects heal target
+                context.player1.clickCard(context.fifthBrother);
+                const battleDroids = context.player1.findCardsByName('battle-droid');
+
+                // Validates healing
+                expect(context.fifthBrother.getHp()).toBe(4);
```

Comment:
`getHp()` always returns max hp, if you want to check hp - damage need to use `remainingHp`